docs/src/main/sphinx/sql/match-recognize.rst

```diff
@@ -612,6 +612,111 @@ This is incorrect::
 
     LAST(A.totalprice + B.totalprice)
 
+aggregate functions
+^^^^^^^^^^^^^^^^^^^
+It is allowed to use aggregate functions in row pattern recognition context.
+They are evaluated over all rows of the current match or over a subset of rows
+based on the matched pattern variables. The
+:ref:`running and final semantics<running_and_final>` are supported,
+``running`` being the default.
+
+The following expression returns the average value of the ``totalprice`` column
+for all rows matched to pattern variable ``A``::
+
+    avg(A.totalprice)
+
+The following expression returns the average value of the ``totalprice`` column
+for all rows matched to pattern variables from subset ``U``::
+
+    avg(U.totalprice)
+
+The following expression returns the average value of the ``totalprice`` column
+for all rows of the match::
+
+    avg(totalprice)
+
+aggregation arguments
+"""""""""""""""""""""
+
+In case when the aggregate function has multiple arguments, it is required that
+all arguments refer consistently to the same set of rows::
+
+    max_by(totalprice, tax) /* aggregate over all rows of the match */
+
+    max_by(CLASSIFIER(A), A.tax) /* aggregate over all rows matched to A */
+
+This is incorrect::
+
+    max_by(A.totalprice, tax)
+
+    max_by(A.totalprice, A.tax + B.tax)
+
+If an aggregate argument does not contain any column reference or
+``classifier`` function, it does not refer to any pattern variable. In such
+case, for the first argument, the universal row pattern variable is implicit,
```

Comment:
In such a case the universal row pattern variable is implicit for the first argument. This means that the aggregate ...